pkg/koordlet/util/system/resctrl.go

```diff
@@ -103,18 +125,33 @@ func IsSupportResctrl() (bool, error) {
 			return false, err
 		}
 		// Kernel cmdline not set resctrl features does not ensure feature must be disabled.
-		klog.Infof("IsSupportResctrl result, cpuSupport: %v, kernelSupport: %v",
+		klog.V(4).Infof("isResctrlAvailableByKernelCmd result, cpuSupport: %v, kernelSupport: %v",
 			cpuSupport, kernelCmdSupport)
 		isSupportResctrl = cpuSupport
 		isInit = true
 	}
 	return isSupportResctrl, nil
 }
 
+func IsSupportResctrlCollector() (bool, error) {
+	var err error
+	collectorOnceFunc.Do(func() {
+		path := GetCPUInfoPath()
+		mbm, err1 := isResctrlMBMAvailableByCpuInfo(path)
+		cqm, err2 := isResctrlCQMAvailableByCpuInfo(path)
+		isSupportResctrlCollector = mbm && cqm
+		err = multierr.Append(err1, err2)
+	})
+	return isSupportResctrlCollector, err
```

Comment:
If IsSupportResctrlCollector is invoked more than once, the err becomes nil.

Reply:
> If IsSupportResctrlCollector is invoked more than once, the err becomes nil.

Yeah, so print errors during initialization, then just use the results later on.